Jolt/Core/Core.h

```diff
@@ -240,7 +245,7 @@ static_assert(sizeof(uint8) == 1, "Invalid size of uint8");
 static_assert(sizeof(uint16) == 2, "Invalid size of uint16");
 static_assert(sizeof(uint32) == 4, "Invalid size of uint32");
 static_assert(sizeof(uint64) == 8, "Invalid size of uint64");
-static_assert(sizeof(void *) == 8, "Invalid size of pointer");
+static_assert(sizeof(void *) == 8 || sizeof(void *) == 4, "Invalid size of pointer" );
```

Comment:
This can then be #if JPH_CPU_ADDRESS_BITS == 32 check that pointer size is 4, else check that it is 8